File: Frontend/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/"

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(private http: HttpClient) { }

  postIncome(incomeDTO: any): Observable<any> {
    return this.http.post(BASIC_URL + "api/income", incomeDTO);
  }

  getAllIncomes(): Observable<any> {
    return this.http.get(BASIC_URL + "api/income/all");
  }

  getIncomeById(id: Number): Observable<any> {
    return this.http.get(BASIC_URL + `api/income/${id}`);
  }

  updateIncome(id: Number, incomeDTO: any): Observable<any> {
    return this.http.put(BASIC_URL + `api/income/${id}`, incomeDTO);
  }

  deleteIncome(id: Number): Observable<any> {
    return this.http.delete(BASIC_URL + `api/income/${id}`);
  }

}

File: expenseTracker/src/main/java/com/example/expenseTracker/services/IncomeService.java
package com.example.expenseTracker.services;

import java.util.List;

import org.springframework.stereotype.Service;

import com.example.expenseTracker.dto.IncomeDTO;
import com.example.expenseTracker.models.Income;

@Service
public interface IncomeService {
	
	Income postIncome(IncomeDTO incomeDTO);
	List<IncomeDTO> getAllIncomes();
	Income updateIncome(Long id,IncomeDTO incomeDTO);
	IncomeDTO getIncomeById(Long id);
	void deleteIncome(Long id);
}
